test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -491,7 +491,8 @@ public final void testFetchMany() throws IOException {
             int count = between(2, 10);
             List<Object> values = new ArrayList<>(count);
             while (values.size() < count) {
-                values.add(generateRandomInputValue(ft));
+                // values should be unique since we also get de-duplication fetching from docvalues
+                values.add(randomValueOtherThanMany(v -> values.contains(v), () -> generateRandomInputValue(ft)));
```

Comment:
I'm not entirely sure if it is better to fix this in the setup or in the assertion in the end. We might also de-dup the results from the native fetchers before comparing, wdyt?